components/FileMenu.vue

```diff
@@ -150,6 +141,28 @@ const closeModal = () => {
 };
 
 watch(errorMessages, openModal);
+
+const handleMenuPosition = () => {
+  const windowInnerHeight = window.innerHeight;
+  const rectBottom = root.value?.getBoundingClientRect().bottom || 0;
+  const bottomDistance = windowInnerHeight - rectBottom;
+  if (bottomDistance < 200) {
+    menuListPosition.value = "top";
+  } else {
+    menuListPosition.value = "bottom";
+  }
+};
+
+onMounted(() => {
+  handleMenuPosition();
+  const fileListElement = document.querySelector(".files__list");
+  fileListElement?.addEventListener("scroll", handleMenuPosition);
+});
+
+onUnmounted(() => {
+  const fileListElement = document.querySelector(".files__list");
+  fileListElement?.removeEventListener("scroll", handleMenuPosition);
+});
```

Comment:
This is to dynamically position single file menu depending on the position of file on the page. 
There is a repetition of:
 const fileListElement = document.querySelector(".files__list");
but since it is element in the parent component (2 components up) I think I cannot have it here as a ref - cuz ref is an element in the current component

Reply:
what's the reason to do the checking for `.files__list` element instead of `root`div here? maybe I can't see something 🤔

Reply:
Cuz it didn't work with root nor window nor document. I found that the scroll event occurs when an element's scrollbar is being scrolled. So here the scrollbars is in .files__list element - so this is the scrollable element here.

Reply:
Thank you for a tip with that IntersectionObserver, it is really cool :). 

But in this case it turned out it is not enough in this case. The intersection observer callback - fires only once, so it is good to position the menu when the file is displayed. But when scrolling - the position didn't change ;(.

So I still have scroll listener but with some improvements:
- it will fire only for files that were already displayed according to the intersection observer -> wasFileDisplayed ref;
- there is some custom throttling of the scroll event, so it doesn't fire so often.

Reply:
Ok so another change in the intersection observer - your settings were the best, additionally I added unobserve :)